app/models/user.rb

```diff
@@ -1564,6 +1638,24 @@ def has_status?
     user_status && !user_status.expired?
   end
 
+  def self.redesigned_menu_ids
+    Discourse.redis.keys("redesigned_user_menu_for_user_*").map do |key|
```

Comment:
Do note that we should avoid using `KEYS` in production. See the warning in https://redis.io/commands/keys/. We probably want to use a `SET` here.